runtime/sema/check_composite_declaration.go

```diff
@@ -405,9 +497,26 @@ func (checker *Checker) declareNestedDeclarations(
 		nestedCompositeTypes = append(nestedCompositeTypes, nestedCompositeType)
 	}
 
+	// Declare nested attachments
+
+	for _, nestedDeclaration := range nestedAttachmentDeclaration {
+		if _, exists := nestedDeclarations[nestedDeclaration.Identifier.Identifier]; !exists {
+			nestedDeclarations[nestedDeclaration.Identifier.Identifier] = nestedDeclaration
+		}
+
+		nestedCompositeType := checker.declareAttachmentType(nestedDeclaration)
+		nestedCompositeTypes = append(nestedCompositeTypes, nestedCompositeType)
+	}
+
 	return
 }
 
+func (checker *Checker) declareAttachmentType(declaration *ast.AttachmentDeclaration) *CompositeType {
+	composite := checker.declareCompositeType(checker.attachmentAsComposite(declaration))
+	composite.baseType = checker.convertNominalType(declaration.BaseType)
```

Comment:
How are we ensuring that the base type of the attachment declaration is always a nominal type?

Reply:
This is enforced in the parser